packages/protocol/test/stability/exchange.ts

```diff
@@ -626,6 +626,81 @@ contract('Exchange', (accounts: string[]) => {
+              setMaxStableBucketFractionTx = await exchange.setMaxStableBucketFraction(
+                toFixed(1 / 22)
+              )
+
+              // Make the capp trigger only after 1 millon coins
+              minSupplyForStableBucketCapTx = await exchange.setMinSupplyForStableBucketCap(
+                new BigNumber('1e24')
+              ) // Todo test playing with this variable
+
+              await mockSortedOracles.setMedianRate(
+                stableToken.address,
+                new BigNumber('0x20000000000000000')
+              ) // 1 CELO = 2 USD
+
+              await stableToken.mint(user, new BigNumber('61e24')) // 6M
+              await registry.setAddressFor(CeloContractName.Exchange, exchange.address)
+            })
+
+            it('has the right maxStableBucketFraction', async () => {
+              assertEqualBN(await exchange.maxStableBucketFraction(), toFixed(1 / 22))
+            })
+
+            it("stable bucket doesn't hit the cap and it shouldn't be resized", async () => {
+              const [tradeableGold, mintableStable] = await exchange.getBuyAndSellBuckets(false)
+              assertEqualBN(mintableStable, new BigNumber('2e21'))
+              assertEqualBN(tradeableGold, new BigNumber('1e21'))
+            })
+
+            it('emits MaxStableBucketFractionSet', async () => {
+              const exchangeLogs = setMaxStableBucketFractionTx.logs.filter(
+                (x) => x.event === 'MaxStableBucketFractionSet'
+              )
+              assert(exchangeLogs.length === 1, 'Did not receive event')
+            })
+
+            it('emits minSupplyForStableBucketCapSet', async () => {
+              const exchangeLogs = minSupplyForStableBucketCapTx.logs.filter(
+                (x) => x.event === 'minSupplyForStableBucketCapSet'
+              )
+              assert(exchangeLogs.length === 1, 'Did not receive event')
+            })
+
+            it('has the correct getStableBucketTokenCap', async () => {
+              assertEqualBN(
+                await exchange.getStableBucketTokenCap(),
+                new BigNumber('2772727272727272816000000')
+              )
+            })
+
+            it('stable bucket hit the cap and it should be resized', async () => {
+              await mockSortedOracles.setMedianRate(
+                stableToken.address,
+                new BigNumber('0x20000000000000000').multipliedBy(2000)
+              ) // bump the price by 10000
+
+              const [tradeableGold, mintableStable] = await exchange.getBuyAndSellBuckets(false)
+
+              console.log(
+                'minteable token ' + new BigNumber(mintableStable.dividedBy(tradeableGold))
+              )
+
+              // ratio between the bucket's shouldn't change
+              assertEqualBN(
+                new BigNumber(mintableStable.dividedBy(tradeableGold)),
+                new BigNumber('4000')
+              )
+            })
+          })
+
```

Comment:
Please add tests that check we are reverting wrong inputs

Reply:
👍